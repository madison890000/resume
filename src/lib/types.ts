export enum Degree {
    Bachelor = 'Bachelor',
    Master = 'Master',
    PhD = 'PhD',
}

export type Gender = 'male' | 'female';


export type Achievement = {
    text: string;
    categories: string[];
};
export type SolutionsOfHowToImplement = string;

export enum JobPositionLevel {
    Staff = 'Staff',
    Senior = 'Senior',
    Middle = '',
    Junior = 'Junior',
}

export type JobPosition =
    'Software Engineer'
    | 'Front-End Engineer'
    | 'Back-End Engineer'
    | 'Full-Stack Engineer';

export type JobResponsibility = string;
export type FEUILib = 'Material UI' | 'antd';
export type ReactFramework = 'NextJs' | 'Create React App' | 'umi';
export type Framework = 'React' | 'Vue' | 'Angular' | 'jQuery';
export type CodeLanguage = 'JAVA' | 'C#' | 'JavaScript';
export type OtherSkillLabels = 'NodeJs' | 'HTML5' | 'CSS3' | 'Webpack';
export type WayOfDevelopment = 'Agile' | 'Waterfall';

export type Keyword =
    Framework
    & CodeLanguage
    & OtherSkillLabels
    & FEUILib
    & ReactFramework
    & WayOfDevelopment;

export enum SkillLevel {
    understand = 'understand',
    familiar = 'familiar',
    proficient = 'proficient',
}

export enum Importance {
    important,
    normal,
    less,
}
export enum CompanyType {
    Enterprise,
    Startup,
}
