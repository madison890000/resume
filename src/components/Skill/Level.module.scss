.level {
  & > span {
    display: inline-block;
    margin: var(--base-gap-unit);
    width: 36px;
    height: 20px;
    background: gray;
  }

  @media screen and (max-width: 414px) {
    & > span {
      width: 20px;
      height: 12px;
    }
  }
}

.levelName {
  font-size: var(--base-font-size-small);
  margin-left: var(--base-gap-unit);
}


.understand {
  & > span:nth-child(2) {
    background: transparent;
  }

  & > span:nth-child(3) {
    background: transparent;
  }
}

.familiar {
  & > span:nth-child(3) {
    background: transparent;
  }
}

.proficient {
  color: black;
}
