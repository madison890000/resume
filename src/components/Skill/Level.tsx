import React from 'react';
import styles from './Level.module.scss';
import classNames from 'classnames';

const LevelUnderstand = () =>
    <div className={classNames(styles.level, styles.understand)}>
        <span />
        <span />
        <span />
        <div className={styles.levelName}>Understand</div>
    </div>
const LevelFamiliar = () =>
    <div className={classNames(styles.level, styles.familiar)}>
        <span />
        <span />
        <span />
        <div className={styles.levelName}>Familiar</div>
    </div>
const LevelProficient = () =>
    <div className={classNames(styles.level, styles.proficient)}>
        <span />
        <span />
        <span />
        <div className={styles.levelName}>Proficient</div>
    </div>

const Level = () => null

Level.Understand = LevelUnderstand;
Level.Familiar = LevelFamiliar;
Level.Proficient = LevelProficient;

export default Level
