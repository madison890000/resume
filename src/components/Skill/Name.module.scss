.important {
  font-weight: bold;
}

.normal {
  color: gray;
  font-weight: bold;
}

.less {
  color: gray;
}
