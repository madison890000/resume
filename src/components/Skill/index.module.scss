.skill {
  display: flex;
  align-items: center;
  justify-content: space-between;

  & > span {
    margin: var(--inline-block-margin);
  }
}

.subTitle {
  font-size: var(--base-font-size-small);
  margin-left: var(--base-gap-unit);
  font-style: italic;
}
