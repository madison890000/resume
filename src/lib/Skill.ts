import { Importance, SkillLevel } from './types';


export default class Skill {
    public name: string;
    public level: SkillLevel;
    public importance: Importance;
    public ages: number;

    constructor(name: string, level: SkillLevel, ages: number, importance: Importance) {
        this.name = name;
        this.level = level;
        this.ages = ages;
        this.importance = importance;
    }

}
