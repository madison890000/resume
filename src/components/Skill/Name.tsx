import React, { PropsWithChildren } from 'react';
import styles from './Name.module.scss';
import { Importance } from '../../lib/types';

const Name: React.FC<PropsWithChildren<{ level: Importance }>> = ({ level, children }) => {
    switch (level) {
        case Importance.important:
            return (
                <div className={styles.important}>{children}</div>
            );
        case Importance.normal:
            return (
                <div className={styles.normal}>{children}</div>
            );
        case Importance.less:
            return (
                <div className={styles.less}>{children}</div>
            );
        default:
            return null
    }
}

export default Name;
