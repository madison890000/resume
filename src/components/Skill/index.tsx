import React from 'react';
import styles from './index.module.scss';
import { Importance, SkillLevel } from '../../lib/types';
import Name from './Name';
import Level from './Level';

interface SkillProps {
    name: string;
    level: SkillLevel;
    ages: number;
    importance: Importance;
}

const Skill = ({ level, name, ages, importance }: SkillProps) => {
    return (
        <div className={styles.skill}>
            <div>
                <Name level={importance}>{name}</Name>
                <div className={styles.subTitle}>{ages} years</div>
            </div>
            {level === SkillLevel.familiar && <Level.Familiar />}
            {level === SkillLevel.proficient && <Level.Proficient />}
            {level === SkillLevel.understand && <Level.Understand />}
        </div>
    );
}

export default Skill;
