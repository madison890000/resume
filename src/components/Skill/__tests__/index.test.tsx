import React from 'react';
import Skill from '../index';
import renderer from 'react-test-renderer';
import { Importance, SkillLevel } from '../../../lib/types';

describe('renders Skill ', () => {

    test('Skill --->level = understand', () => {
        const SkillUnderstandNormal = renderer.create(
            <Skill
                name="understand normal Skill"
                ages={5}
                level={SkillLevel.understand}
                importance={Importance.normal}
            />
        )
            .toJSON();
        expect(SkillUnderstandNormal)
            .toMatchSnapshot();
        const SkillUnderstandLess = renderer.create(
            <Skill
                name="understand less Skill"
                ages={5}
                level={SkillLevel.understand}
                importance={Importance.less}
            />
        )
            .toJSON();
        expect(SkillUnderstandLess)
            .toMatchSnapshot();
        const SkillUnderstandImportant = renderer.create(
            <Skill
                name="understand important Skill"
                ages={5}
                level={SkillLevel.understand}
                importance={Importance.important}
            />
        )
            .toJSON();
        expect(SkillUnderstandImportant)
            .toMatchSnapshot();
    });

    test('Skill --->level = familiar', () => {
        const SkillFamiliarNormal = renderer.create(
            <Skill
                name="familiar normal Skill"
                ages={5}
                level={SkillLevel.familiar}
                importance={Importance.normal}
            />
        )
            .toJSON();
        expect(SkillFamiliarNormal)
            .toMatchSnapshot();
        const SkillFamiliarLess = renderer.create(
            <Skill
                name="familiar less Skill"
                ages={5}
                level={SkillLevel.familiar}
                importance={Importance.less}
            />
        )
            .toJSON();
        expect(SkillFamiliarLess)
            .toMatchSnapshot();
        const SkillFamiliarImportant = renderer.create(
            <Skill
                name="familiar important Skill"
                ages={5}
                level={SkillLevel.familiar}
                importance={Importance.important}
            />
        )
            .toJSON();
        expect(SkillFamiliarImportant)
            .toMatchSnapshot();
    });

    test('Skill --->level = proficient', () => {
        const SkillProficientNormal = renderer.create(
            <Skill
                name="proficient normal Skill"
                ages={5}
                level={SkillLevel.proficient}
                importance={Importance.normal}
            />
        )
            .toJSON();
        expect(SkillProficientNormal)
            .toMatchSnapshot();
        const SkillProficientLess = renderer.create(
            <Skill
                name="proficient less Skill"
                ages={5}
                level={SkillLevel.understand}
                importance={Importance.less}
            />
        )
            .toJSON();
        expect(SkillProficientLess)
            .toMatchSnapshot();
        const SkillProficientImportant = renderer.create(
            <Skill
                name="proficient important Skill"
                ages={5}
                level={SkillLevel.understand}
                importance={Importance.important}
            />
        )
            .toJSON();
        expect(SkillProficientImportant)
            .toMatchSnapshot();
    });

})
