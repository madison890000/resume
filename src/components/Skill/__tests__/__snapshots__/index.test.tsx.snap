// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Skill  Skill --->level = familiar 1`] = `
<div
  className="skill"
>
  <div>
    <div
      className="normal"
    >
      familiar normal Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level familiar"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Familiar
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = familiar 2`] = `
<div
  className="skill"
>
  <div>
    <div
      className="less"
    >
      familiar less Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level familiar"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Familiar
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = familiar 3`] = `
<div
  className="skill"
>
  <div>
    <div
      className="important"
    >
      familiar important Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level familiar"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Familiar
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = proficient 1`] = `
<div
  className="skill"
>
  <div>
    <div
      className="normal"
    >
      proficient normal Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level proficient"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Proficient
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = proficient 2`] = `
<div
  className="skill"
>
  <div>
    <div
      className="less"
    >
      proficient less Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level understand"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = proficient 3`] = `
<div
  className="skill"
>
  <div>
    <div
      className="important"
    >
      proficient important Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level understand"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = understand 1`] = `
<div
  className="skill"
>
  <div>
    <div
      className="normal"
    >
      understand normal Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level understand"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = understand 2`] = `
<div
  className="skill"
>
  <div>
    <div
      className="less"
    >
      understand less Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level understand"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;

exports[`renders Skill  Skill --->level = understand 3`] = `
<div
  className="skill"
>
  <div>
    <div
      className="important"
    >
      understand important Skill
    </div>
    <div
      className="subTitle"
    >
      5
       years
    </div>
  </div>
  <div
    className="level understand"
  >
    <span />
    <span />
    <span />
    <div
      className="levelName"
    >
      Understand
    </div>
  </div>
</div>
`;
